packages/grid/_modules_/grid/hooks/features/pagination/useGridPagination.ts

```diff
@@ -6,94 +6,121 @@ import { GridPageChangeParams } from '../../../models/params/gridPageChangeParam
 import { useGridApiOptionHandler } from '../../root/useGridApiEventHandler';
 import { useGridApiMethod } from '../../root/useGridApiMethod';
 import { optionsSelector } from '../../utils/optionsSelector';
+import { useGridState } from '../core/useGridState';
 import { gridContainerSizesSelector } from '../../root/gridContainerSizesSelector';
 import { useLogger } from '../../utils/useLogger';
-import { useGridReducer } from '../core/useGridReducer';
 import { useGridSelector } from '../core/useGridSelector';
 import { visibleGridRowCountSelector } from '../filter/gridFilterSelector';
-import {
-  GRID_INITIAL_PAGINATION_STATE,
-  GridPaginationActions,
-  gridPaginationReducer,
-  GridPaginationState,
-  setGridPageActionCreator,
-  setGridPageSizeActionCreator,
-  setGridPaginationModeActionCreator,
-  setGridRowCountActionCreator,
-} from './gridPaginationReducer';
+import { GridPaginationState } from './gridPaginationState';
 
-const PAGINATION_STATE_ID = 'pagination';
+function computeState(state) {
+  const pageCount =
+    state.pageSize && state.rowCount > 0 ? Math.ceil(state.rowCount / state.pageSize!) : 1;
+
+  return {
+    ...state,
+    pageCount,
+    page: Math.min(
+      pageCount - 1,
+      state.controlledPage !== null ? state.controlledPage : state.page,
```

Comment:
I think that `controlledPage`  should not be part of the state, how about you add the props as a 2nd arg of this function